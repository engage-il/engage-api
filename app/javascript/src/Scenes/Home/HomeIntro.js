// @flow
import * as React from 'react'
import styled from 'react-emotion'
import { Link, CloseButton as CloseButton$, Defer } from '@/components'
import { local } from '@/storage'
import { mixins } from '@/styles'

type State = {
  isAccepted: boolean
}

export class HomeIntro extends React.Component<{}, State> {
  state = {
    isAccepted: !!local.get('intro-cleared')
  }

  // events
  didClickAccept = () => {
    this.setState({ isAccepted: true }, () => {
      local.set('intro-cleared', 'true')
    })
  }

  // lifecycle
  componentDidMount () {
    local.set('intro-visited', 'true')
  }

  render () {
    const { isAccepted } = this.state

    return (
      <Defer>
        {!isAccepted && (
          <Intro2>
            <h2>It's time the government went digital.</h2>
            <h4>You should be heard by your representatives, and you should know how they represent you.</h4>
            <h4><Link to='#'>See the bills</Link> that impact your life and get in touch with your congress people today.</h4>
            <CloseButton
              onClick={this.didClickAccept}
            />
          </Intro2>
        )}
      </Defer>
    )
  }
}

const Intro2 = styled.div`
  ${mixins.pageContent};

  align-self: center;
  padding-top: 70px;
  padding-bottom: 70px;

  > h2 {
    margin-bottom: 25px;
  }

  > h4 + h4 {
    margin-top: 15px;
  }
`

const CloseButton = styled(CloseButton$)`
  position: absolute;
  top: 30px;
  right: 30px;
`
