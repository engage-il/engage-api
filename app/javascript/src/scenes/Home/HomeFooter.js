// @flow
import * as React from 'react'
import styled from 'react-emotion'
import { Footer } from '@/components'
import { mixins, colors } from '@/styles'
import { quill } from '&/images'

const size = {
  width: 77,
  height: 25
}

export function HomeFooter () {
  return (
    <Footer>
      <Background>
        <Content>
          <img
            src={quill}
            alt='Quill Logo'
            style={size}
            {...size}
          />
          <h3>Non-partisan, educational, accessible</h3>
          <p>Legislated is a non-profit organization started at Chi Hack Night by a dedicated group of volunteers.</p>
        </Content>
      </Background>
    </Footer>
  )
}

const Background = styled.div`
  ${mixins.flexRow};
  ${mixins.flexCenter};

  padding: 50px 25px;
  background-color: ${colors.secondary};
`

const Content = styled.div`
  ${mixins.flexColumn};

  align-items: center;
  max-width: 550px;
  color: ${colors.white};
  text-align: center;

  > * + * {
    margin-top: 15px;
  }
`
