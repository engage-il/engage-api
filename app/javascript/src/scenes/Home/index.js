// @flow
export {
  HomeRenderer as Home
} from './Home'
