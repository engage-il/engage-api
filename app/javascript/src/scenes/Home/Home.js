// @flow
import * as React from 'react'
import { graphql } from 'react-relay'
import { withRouter } from 'react-router-dom'
import type { ContextRouter } from 'react-router-dom'
import styled from 'react-emotion'
import { HomeIntro } from './HomeIntro'
import { RelayRenderer, BillSearch, Button } from '@/components'
import type { Viewer, SearchParams } from '@/types'
import { mixins } from '@/styles'

const DEFAULT_PARAMS = {
  query: ''
}

type Props = {
  viewer: ?Viewer
} & ContextRouter

let Home = class Home extends React.Component<*, Props, *> {
  params: SearchParams = DEFAULT_PARAMS

  // events
  didChangeParams = (params: SearchParams) => {
    this.params = params
  }

  didClickViewAll = () => {
    // it would be better to pass params / refresh the view
    // by updating the search query in the url
    const { history } = this.props
    history.push({
      pathname: '/bills',
      state: {
        params: this.params
      }
    })
  }

  // lifecycle
  render () {
    const { viewer } = this.props

    return (
      <Scene>
        <HomeIntro />
        <BillSearch
          viewer={viewer}
          onFilter={this.didChangeParams}
        />
        <BillsButton
          onClick={this.didClickViewAll}
          isSecondary
          children='View All Bills'
        />
      </Scene>
    )
  }
}

const Scene = styled.section`
  ${mixins.flexColumn};

  position: relative;
  align-self: center;
`

const BillsButton = styled(Button)`
  align-self: center;
  margin-bottom: 70px;
`

Home = withRouter(Home)

export { Home }

export function HomeRenderer () {
  const query = graphql`
    query HomeQuery(
      $filter: BillsSearchFilter!,
      $count: Int!,
      $cursor: String!
    ) {
      viewer {
        ...BillSearch_viewer
      }
    }
  `

  return (
    <RelayRenderer
      root={Home}
      query={query}
      getVariables={() => ({
        count: 3,
        cursor: '',
        filter: {
          key: 'home',
          ...DEFAULT_PARAMS
        }
      })}
    />
  )
}
