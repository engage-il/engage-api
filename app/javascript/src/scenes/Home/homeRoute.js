// @flow
import React from 'react'
import { graphql } from 'react-relay'
import { Home } from './Home'
import type { RelayRouteConfig } from '@/types'

export const homeRoute: RelayRouteConfig = {
  query: graphql`
    query homeRouteQuery(
      $filter: BillsSearchFilter!,
      $count: Int!,
      $cursor: String!
    ) {
      viewer {
        ...HomeScene_viewer
      }
    }
  `,
  getInitialVariables (props) {
    return {
      count: 3,
      cursor: '',
      filter: {
        query: ''
      }
    }
  },
  render (props) {
    if (props) {
      return <Home {...props} />
    } else {
      return <Home viewer={null} />
    }
  }
}
