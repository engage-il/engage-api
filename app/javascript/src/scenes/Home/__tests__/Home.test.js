/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { defaultsDeep } from 'lodash'
import { Home } from '../Home'

// subject
let subject

const defaults = {
  viewer: 'test-viewer'
}

function loadSubject (props = {}) {
  subject = shallow(<Home {...defaultsDeep(props, defaults)} />)
}

// specs
afterEach(() => {
  subject = null
})

describe('#render', () => {
  it('shows the scene', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })
})
