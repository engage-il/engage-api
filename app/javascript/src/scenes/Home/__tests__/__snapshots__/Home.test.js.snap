// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render shows the scene 1`] = `
<Styled(section)>
  <HomeIntro />
  <BillSearch
    onFilter={[Function]}
    viewer="test-viewer"
  />
  <Styled(Button)
    isSecondary={true}
    to={
      Object {
        "pathname": "/bills",
        "state": Object {
          "params": undefined,
        },
      }
    }
  >
    View All Bills
  </Styled(Button)>
</Styled(section)>
`;
